### DISCUSSION TITLE
Export long axis data

### DISCUSSION THREAD

**Discussion by reepaes** [mtex-toolbox/mtex#2038] [Ask Anything]
Hi, I would like to know how I can export the long-axis orientation data
I follow the script available  at https://mtex-toolbox.github.io/EllipseBasedParameters.html to find the grain's long axis
Then I tried without success to obtain the orientation value due to:

%% Geting the long axis orientations
grains,grains.longAxis=rose
rose.export('myfile.txt',degree');
 
Could you help me please?
Best
Renata

**Comment by zmichels** [comment]
Hi,In what format/coordinates do you want to export the axes/vector data? ZachOn Jan 15, 2024, at 9:50 AM, reepaes ***@***.***> wrote:﻿
Hi, I would like to know how I can export the long-axis orientation data
I follow the script available  at https://mtex-toolbox.github.io/EllipseBasedParameters.html to find the grain's long axis
Then I tried without success to obtain the orientation value due to:
%% Geting the long axis orientations
grains,grains.longAxis=rose
rose.export('myfile.txt',degree');
Could you help me please?
Best
Renata

**Reply by reepaes** [reply]
I would like to export in polar angle to .txt format
Thanks

**Reply by zmichels** [reply]
Hi,

I think the code below does what you want for export. It also plots the data as a visual sanity check of the angles you are exporting.

Cheers,
Zach

%% Grain long axes computation and export
%% Smooth grains
g = smooth(g,5,'moveTriplePoints','moveOuterBoundary');

%% Get some ellipse data fit to the smothed grain hulls
% fit ellipses to grain hulls
[omega,longAxLength,shortAxLength] = fitEllipse(g);

%% polarhistogram of long-axis angle to X-direction (omega)
figure,
polarhistogram([omega omega+pi],360/10,'Normalization','pdf','FaceColor',[.5 .5 .5],'EdgeColor','k');
hgexport(gcf,'longAxes_polarHistogram.png', hgexport('factorystyle'), 'Format', 'png')

%% Create a table with the data and variable names
T = table(omega, longAxLength, shortAxLength, 'VariableNames', { 'angle', 'long_Axis_Length', 'short_axis_length'});

%% Export table of ellipse-fit data 
writetable(T, 'MyFile.txt')

**Reply by reepaes** [reply]
Thank you, it worked!

**Reply by reepaes** [reply]
@zmichels Could you help me with just one more Doubt; sorry to ask so many questions.
How can I have the angles in degrees and azimuth coordinates?
I made this way but I just got it in degrees but I think is still in polar. 

%% %% reconstruct grains and smooth them
[grains, ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',5*degree);
ebsd(grains(grains.grainSize<10)) = [];
[grains, ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',5*degree);
grains(grains.isBoundary) = [];

grains=smooth(grains('indexed'),10,'moveTriplePoints');

% plot the grains
plot(grains,'micronbar','off','lineWidth',2)
%% Fit Ellipses
[omega,a,b] = grains.fitEllipse;

plotEllipse(grains.centroid,a,b,omega,'lineColor','w','linewidth',2)
%% Long Axes
% visualize the aspect ratio
plot(grains,grains.aspectRatio,'linewidth',2,'micronbar','off')
setColorRange([0,4])
mtexColorbar('title','aspect ratio')

% and on top the long axes
hold on
quiver(grains,grains.longAxis,'Color','white')
hold off
%% Shape perfered orientation Long Axis Distribution
numBin = 50;
subplot(1,2,1)
weights = grains('Quartz-new').area .* (grains('Quartz-new').aspectRatio-1);
histogram(grains('Quartz-new').longAxis,numBin, 'weights', weights)
title('Quartz Long Axis')
%% Create a table with the data and variable names
T = table(omega*180/pi, a, b, 'VariableNames', { 'angle', 'a', 'b'});

%% Export table of ellipse-fit data
writetable(T, 'MyFile_data5.txt')

I tried also to do ''rho.omega'' but didn't work it. 
Thank you
